core/src/test/java/io/confluent/kafka/schemaregistry/SASLClusterTestHarness.java

```diff
@@ -0,0 +1,182 @@
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry;
+
+import kafka.security.minikdc.MiniKdc;
+import kafka.server.KafkaConfig;
+import kafka.utils.JaasTestUtils;
+import kafka.utils.TestUtils;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.protocol.SecurityProtocol;
+import org.apache.kafka.common.security.authenticator.LoginManager;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+import scala.collection.Seq;
+
+import javax.security.auth.login.Configuration;
+import java.io.File;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Properties;
+
+// sets up SASL for ZooKeeper and Kafka.
+public class SASLClusterTestHarness extends ClusterTestHarness {
+  public static final String JAAS_CONF = "java.security.auth.login.config";
+  public static final String KRB5_CONF = "java.security.krb5.conf";
+  public static final String ZK_AUTH_PROVIDER = "zookeeper.authProvider.1";
+
+  private static MiniKdc kdc;
+  private static File kdcHome;
+  private static String krb5ConfPath;
+  private static File jaasFile;
+
+  private static final Logger log = LoggerFactory.getLogger(SASLClusterTestHarness.class);
+
+  public SASLClusterTestHarness() {
+    super(DEFAULT_NUM_BROKERS);
+  }
+
+  @Override
+  protected SecurityProtocol getSecurityProtocol() {
+    return SecurityProtocol.SASL_PLAINTEXT;
+  }
+
+  @BeforeClass
+  public static void setUpKdc() throws Exception {
+    destroySaslHelper();
+    File zkServerKeytab = File.createTempFile("zookeeper-", ".keytab");
+    File kafkaKeytab = File.createTempFile("kafka-", ".keytab");
+    File schemaRegistryKeytab = File.createTempFile("schema-registry-", ".keytab");
+
+    // build and write the JAAS file.
+    JaasTestUtils.JaasSection serverSection = createJaasSection(zkServerKeytab,
+            "zookeeper/localhost@EXAMPLE.COM", "Server", "zookeeper");
+    // NOTE: there is only one `Client` section in the Jaas configuraiton file. Both the internal embedded Kafka
+    // cluster and the schema registry share the same principal. This is required because within the same JVM (eg
+    // these tests) one cannot have two sections, each with its own ZooKeeper client SASL credentials.
+    JaasTestUtils.JaasSection clientSection = createJaasSection(kafkaKeytab,
+            "kafka/localhost@EXAMPLE.COM", "Client", "zookeeper");
+    JaasTestUtils.JaasSection kafkaServerSection = createJaasSection(kafkaKeytab,
+            "kafka/localhost@EXAMPLE.COM", "KafkaServer", "kafka");
+    JaasTestUtils.JaasSection kafkaClientSection = createJaasSection(schemaRegistryKeytab,
+            "schema-registry/localhost@EXAMPLE.COM", "KafkaClient", "kafka");
+    jaasFile = File.createTempFile("schema_registry_tests", "_jaas.conf");
```

Comment:
It may make sense to us `TestUtils.tempFile` as it configures it to be deleted on exit.